### PR TITLE
Use declarative setuptools

### PR DESCRIPTION
**Pull request by charlesbluca**
Haven't really touched the cython bits here, but this should resolve https://github.com/dask/distributed/issues/8369

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -10,9 +10,23 @@ build-backend = "setuptools.build_meta"
...
 name = "crick"
+description="High performance approximate and streaming algorithms"
+maintainers = [{name = "Jim Crist", email = "crist042@umn.edu"}]
+license = {text = "BSD-3-Clause"}
+readme = "README.rst"
 requires-python = ">=3.8"
 dynamic = ["version"]
 
+[project.urls]
+Source = "https://github.com/dask/crick"
```
Comment (milesgranger) [inline pyproject.toml:21]: Is `Source` a valid option?
Maybe `Home` (just based from reading this https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#urls)

File: pyproject.toml [hunk 2]
```diff
@@ -10,9 +10,23 @@ build-backend = "setuptools.build_meta"
 
 [project]
 name = "crick"
+description="High performance approximate and streaming algorithms"
+maintainers = [{name = "Jim Crist", email = "crist042@umn.edu"}]
```
Comment (milesgranger) [inline pyproject.toml:14]: I believe Jim [no longer has time/desire to maintain crick](https://github.com/dask/crick/issues/25#issuecomment-1317701130). @fjetter I see dask/dask and distributed both have Matt as the maintainers, do we want to add/change this as well for crick?

File: pyproject.toml (hunk 1 above)
Comment (milesgranger) [reply pyproject.toml:21]: Nvm, I see it's used in dask/distributed.

File: pyproject.toml (hunk 2 above)
Comment (fjetter) [reply pyproject.toml:14]: I suggest to keep this as is for now and fix the packaging without worrying about this too much. I'm fine changing this but I want Jim's ACK and frankly I don't know what else to put in here. We should probably setup an info@dask.org email address if that doesn't exist yet (@jacobtomlinson does something like that exist?)

**Review by milesgranger — APPROVED** [review]

**Pull request merged**